code/ATMOSPHERICS/components/unary/tank.dm

```diff
@@ -83,21 +90,93 @@
 		GAS_OXYGEN, (25*ONE_ATMOSPHERE*O2STANDARD)*(starting_volume)/(R_IDEAL_GAS_EQUATION*air_contents.temperature),
 		GAS_NITROGEN, (25*ONE_ATMOSPHERE*N2STANDARD)*(starting_volume)/(R_IDEAL_GAS_EQUATION*air_contents.temperature))
 
+/obj/machinery/atmospherics/unary/tank/empty
+	icon_state = "air"
+	name = "Pressure Tank"
+	default_colour = "#0000b7"
+	anchored = 0
+
 /obj/machinery/atmospherics/unary/tank/update_icon()
 	..()
 
 /obj/machinery/atmospherics/unary/tank/disconnect(obj/machinery/atmospherics/reference)
 	..()
 	update_icon()
 
+/obj/machinery/atmospherics/unary/tank/verb/rotate()
+	set name = "Rotate Clockwise"
+	set category = "Object"
+	set src in oview(1)
+
+	if(src.anchored || usr:stat)
+		to_chat(usr, "It is fastened to the floor!")
+		return 0
+	src.dir = turn(src.dir, -90)
+	return 1
+
+/obj/machinery/atmospherics/unary/tank/verb/rotate_ccw()
+	set name = "Rotate Counter Clockwise"
+	set category = "Object"
+	set src in oview(1)
+
+	if(src.anchored || usr:stat)
+		to_chat(usr, "It is fastened to the floor!")
+		return 0
+	src.dir = turn(src.dir, 90)
+	return 1
+
+/obj/machinery/atmospherics/unary/tank/wrenchAnchor(var/mob/user)
+	. = ..()
+	if(!.)
+		return
+	if(anchored)
+		verbs -= rotate_verbs
+		initialize_directions = dir
+		initialize()
+		build_network()
+		if (node1)
+			node1.initialize()
+			node1.build_network()
+	else
+		verbs += rotate_verbs
+		if(node1)
+			node1.disconnect(src)
+			node1 = null
+		if(network)
+			qdel(network)
+			network = null
+		initialize_directions = 0 //this prevents things from attaching to us when we're unanchored
+		update_icon()
 
 /obj/machinery/atmospherics/unary/tank/attackby(var/obj/item/weapon/W as obj, var/mob/user as mob)
 	if(istype(W, /obj/item/device/rcd/rpd) || istype(W, /obj/item/device/pipe_painter))
 		return // Coloring pipes.
 	if (istype(W, /obj/item/device/analyzer) && get_dist(user, src) <= 1)
-		user.visible_message("<span class='attack'>[user] has used [W] on [bicon(icon)] [src]</span>", "<span class='attack'>You use \the [W] on [bicon(icon)] [src]</span>")
 		var/obj/item/device/analyzer/analyzer = W
 		user.show_message(analyzer.output_gas_scan(air_contents, src, 0), 1)
+	
+	//deconstruction
+	if(iswelder(W) && !anchored)
+		var/obj/item/weapon/weldingtool/WT = W
+		if(!WT.remove_fuel(0,user))
+			to_chat(user, "<span class='notice'>You need more welding fuel to complete this task.</span>")
+			return
+		playsound(src, 'sound/items/Welder2.ogg', 100, 1)
+		user.visible_message("<span class='notice'>[user] starts disassembling \the [src].</span>", \
+							"<span class='notice'>You start disassembling \the [src].</span>")
+		if(do_after(user, src, 40))
+			user.visible_message("<span class='warning'>[user] dissasembles \the [src].</span>", \
+			"<span class='notice'>You dissasemble \the [src].</span>")
+			getFromPool(/obj/item/stack/sheet/metal, get_turf(src), 5)
+			qdel(src) //the air contents just get deleted
```

Comment:
Currently when you deconstruct these the air contents just get deleted. While it would be hilarious if puncturing the high-pressure tank caused a big explosion and blew off your arms, I decided to just delete the contents because 1) it's less hassle to players not having to drain these when they want to renovate the area, and 2) I'm lazy.

Reply:
You have to puncture canisters to deconstruct them, kind of gay tbh.

Reply:
Adding puncture mechanics would also mean extra spritework and that's not something I'm willing to bother with, especially since these things have 4 recolours that would also need sprites.

Reply:
Actually I guess I could make puncturing them just create a small explosion and spawn the sheets used to create them in the epicenter. Either way it kind of feels like a dumb gotcha mechanic, though it is tempting.